### PR TITLE
chore(docs): updated reference + upgrade guide

### PR DESCRIPTION
**Pull request by eric-horodyski**
Update the "PortalsPlugin" and "Upgrade Guides" documentation to account for changes made in #166. 

Once this is merged, I would like to request a release of `@ionic/portals@0.6.0` 😄

### CODE REVIEW

File: website/docs/reference/web/portals-plugin.md [hunk 1]
```diff
@@ -9,12 +9,12 @@ The PortalsPlugin class is the main way to interface with a Portal instance. It
 
 ### PortalsPlugin
 
-A type definining the `PortalsPlugin` API.
+A type defining the `PortalsPlugin` API.
```
Comment (Steven0351) [inline website/docs/reference/web/portals-plugin.md:12]: Typo crushing for the win 👍

File: website/docs/reference/web/portals-plugin.md [hunk 2]
```diff
@@ -74,18 +74,17 @@ type SubscriptionCallback<T = unknown> = (result: { topic: string, data: T; }) =
...
-    topic: 'dismiss',
-    data: 'cancel',
-}
+type Messages = 
+  | { topic: 'modal:dismiss', data: 'cancel' | 'failure' }
+  | { topic: 'modal:success', data: number };
 
 // Publishes "cancel" to the "dismiss" topic
-Portals.publish(message);
+Portals.publish<Messages>({ topic: 'modal:dismiss', data: 'cancel' });
```
Comment (Steven0351) [inline website/docs/reference/web/portals-plugin.md:87]: Overall I really like the example. Should we also still show other valid uses like `Portals.publish({ topic: 'something:else', data: 'asdf' })`? I don't really have a huge opinion one way or the other.

**Review by carlpoole — APPROVED** [review]

File: website/docs/reference/web/portals-plugin.md (hunk 2 above)
Comment (eric-horodyski) [reply website/docs/reference/web/portals-plugin.md:88]: 👍 FTFY

**Review by Steven0351 — APPROVED** [review]
Perfection

**Review by giralte-ionic — APPROVED** [review]

**Comment by Steven0351** [comment]
We should update the docs for the changes in #169 as well

**Pull request merged**